src/Hashable.php

```diff
@@ -20,15 +20,15 @@ interface Hashable
      *
      * @return mixed
      */
-    function hash();
+    public function hash();
 
     /**
      * Determines if two objects should be considered equal. Both objects will
      * be instances of the same class but may not be the same instance.
      *
-     * @param $obj An instance of the same class to compare to.
+     * @param $obj self An instance of the same class to compare to.
```

Comment:
I don't think this is correct - it could be anything at all. See https://github.com/php-ds/ext-ds/issues/140

Reply:
Man, I am so confused. 

I go down the same path as many of the commenters in that thread... First, I thought `self` is correct, then `object`, then, certainly `?object`, but then you're all saying that strings, ints, floats, what else? resources, arrays and callables should *also* be accepted?? I'm assuming Exceptions, too? lol.

I can understand `?object`, but just passing anything seems like a clear design flaw to me. But I'm very very unfamiliar w/ this project. I just wanted to do some good will and that conversation happened after I did the bulk of the changes.

Reply:
> Man, I am so confused.

:grin: no problem, I was for a while too. The thing here is that `equals` is used to determine equality when an object is used as a key in a hash-based structure. ie. `Map` will internally call an object's `equals` method when looking up that object in the hash table. 

The issue is that you can of course call `equals` directly, ie. `$map->equals(null)`. You can pass anything you want to that method, which means it is responsible for checking that the given arg is an instance of the same class etc. The doc comment should reflect that.

Reply:
Regardless of the correct type, it should precede the parameter name, not come after it.